doc/libghdl/ASTTraversing.rst

```diff
@@ -0,0 +1,12 @@
+.. _LIB:ASTTraversing:
+
+AST Traversing
+##############
+
+If someone wants to use `libghdl`, then he probably wants to know how to traverse the AST to get the information about nodes of design units.
```

Comment:
There are examples of traverse functions in `pyutils.py`.  The function `nodes_iter` is a visitor on all nodes, while `declarations_iter` and `concurrent_stmts_iter` are more selective.